src/components/scripture/dto/scripture-range.validator.ts

```diff
@@ -0,0 +1,21 @@
+import { ValidationArguments } from 'class-validator';
+import { Merge } from 'type-fest';
+import { createValidationDecorator } from '../../../common/validators/validateBy';
+import { scriptureToVerseRange } from '../reference';
+import { ScriptureRange } from './scripture-range.dto';
+
+// We assume this is only used on the ScriptureRange object
+type ValidationArgs = Merge<ValidationArguments, { object: ScriptureRange }>;
+
+export const IsValidOrder = createValidationDecorator({
+  name: 'ScriptureRange',
+  validator: {
+    validate: (_value, { object: range }: ValidationArgs) => {
+      const verseRange = scriptureToVerseRange(range);
+      return verseRange.start <= verseRange.end;
```

Comment:
Does this execute after `ScriptureReferenceInput` is validated? i.e. start & end are, at this point, valid objects?

Reply:
It validates following the below steps
- Validates `start`
- Validates order
- Validates `end` 

And it returns all the error messages like below
```
"errors": {
   "product.scriptureReferences.0.start.verse": {
        "ScriptureVerse": "Genesis 3 does not have verse 1000"
    },
    "product.scriptureReferences.0.end": {
         "ScriptureRange": "Scripture range must end after the starting point"
    }
},
```

Reply:
So while executing this validator, we don't know `start` and `end` are already `validated`

Reply:
That's a problem because `scriptureToVerseRange` needs the range to be valid. Does flipping the order make a difference?
```tsx
  @Field()
  @IsValidOrder()
  @ValidateNested()
  @Type(() => ScriptureReferenceInput)
  end: ScriptureReferenceInput;
```

Reply:
It's the same. 😢

Reply:
Btw `scriptureToVerseRange ` returns `start` and `end` even if the `range` is invalid
```
"start": {
      "book": "Genesis",
      "chapter": 3,
      "verse": 1000
},

result: 
start: 1055
```

But if the chapter is invalid, it returns like this.
```
"start": {
     "book": "Genesis",
     "chapter": 1000,
     "verse": 3
},

result:
start: NaN
```

Reply:
So should I have to validate the `range` in the `IsValidOrder` first and if at least one is failed, it should return `true`?

Reply:
Yeah that code that was copied/ported needs some love. We probably need to check for `NaN` and return true in that case for the same reason. `NaN < x === false`

Reply:
Can you let me know what is `x`? 😉 
Also, it returns a valid `start` number even if the `start` is invalid.
```
"start": {
      "book": "Genesis",
      "chapter": 3,
      "verse": 1000
},

result: 
start: 1055
```

Reply:
So I think we need to validate all the fields one by one using `validateBook`, `validateChapter` and `validateVerse`

Reply:
Yeah I think that would be best for now